Index: oprofile-0.9.6/src/m4/binutils.m4
===================================================================
--- oprofile-0.9.6.orig/src/m4/binutils.m4	2009-11-24 09:25:16.000000000 -0600
+++ oprofile-0.9.6/src/m4/binutils.m4	2016-11-18 13:59:19.709972507 -0600
@@ -27,8 +27,10 @@
 	AC_MSG_CHECKING([whether bfd_get_synthetic_symtab() exists in BFD library])
 	rm -f test-for-synth
 	AC_LANG_CONFTEST(
-		[AC_LANG_PROGRAM([[#include <bfd.h>]],
-			[[asymbol * synthsyms;	bfd * ibfd = 0; 
+		[AC_LANG_PROGRAM([[#include <bfd.h>]
+		[#include <stdio.h]
+		[static bfd _ibfd;]],
+			[[asymbol * synthsyms;	bfd * ibfd = &_ibfd;
 			long synth_count = bfd_get_synthetic_symtab(ibfd, 0, 0, 0, 0, &synthsyms);
 			extern const bfd_target bfd_elf64_powerpc_vec;
 			extern const bfd_target bfd_elf64_powerpcle_vec;
