Index: oprofile-0.9.6/src/daemon/liblegacy/p_module.h
===================================================================
--- oprofile-0.9.6.orig/src/daemon/liblegacy/p_module.h
+++ oprofile-0.9.6/src/daemon/liblegacy/p_module.h
@@ -182,7 +182,8 @@ struct module_info
 #define NEW_MOD_INITIALIZING	64
 
 int sys_init_module(char const * name, const struct module *);
-int query_module(char const * name, int which, void *buf, size_t bufsize,
+int __attribute__((weak))
+query_module(char const * name, int which, void *buf, size_t bufsize,
 		 size_t *ret);
 
 /* Values for query_module's which.  */
